bug: trialing dash breaks routing

Repro: https://github.com/ernado/ogen-trialing-bug
```bash
gh repo clone ernado/ogen-trialing-bug
cd ogen-trialing-bug
go test
```

```
--- FAIL: TestResolve (0.00s)
    router_test.go:16: 
        	Error Trace:	/tmp/ogen-trialing-bug/router_test.go:16
        	Error:      	Not equal: 
        	            	expected: "listClusters"
        	            	actual  : "getCluster"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-listClusters
        	            	+getCluster
        	Test:       	TestResolve
FAIL
exit status 1
FAIL	bug	0.002s
```

```yaml
openapi: 3.0.3
info:
  title: API
  version: 0.1.0
paths:
  /clusters/:
    get:
      operationId: listClusters
      responses:
        default: {$ref: '#/components/responses/Error'}
        '200':
          description: List of clusters
          content:
            application/json:
              schema:
                type: array
                items:
                  $ref: '#/components/schemas/Cluster'
    post:
      operationId: createCluster
      requestBody:
        required: true
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/Cluster'
      responses:
        default: {$ref: '#/components/responses/Error'}
        '200':
          description: Created cluster
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Cluster'
  /clusters/{cluster_id}:
    get:
      operationId: getCluster
      parameters:
        - $ref: '#/components/parameters/ClusterID'
      responses:
        default: {$ref: '#/components/responses/Error'}
        '200':
          description: Cluster
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Cluster'
```

The `getCluster` is called on `/clusters/` URI intstead of `listClusters`. Routing is not generated for `getCluster`.

[This](https://github.com/ernado/ogen-trialing-bug/blob/main/router_test.go) fails:
```go
package bug

import (
	"net/http"
	"testing"

	"github.com/stretchr/testify/require"

	"bug/oas"
)

func TestResolve(t *testing.T) {
	var h oas.Server
	route, ok := h.FindRoute(http.MethodGet, "/clusters/")
	require.True(t, ok)
	require.Equal(t, "listClusters", route.OperationID())
}
```

--- a/examples/ex_k8s/oas_router_gen.go
+++ b/examples/ex_k8s/oas_router_gen.go
@@ -13026,6 +13026,16 @@ func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 					break
 				}
 
+				if len(elem) == 0 {
+					switch r.Method {
+					case "GET":
+						s.handleLogFileListHandlerRequest([0]string{}, elemIsEscaped, w, r)
+					default:
+						s.notAllowed(w, r, "GET")
+					}
+
+					return
+				}
 				// Param: "logpath"
 				// Leaf parameter
 				args[0] = elem

--- a/gen/_template/router.tmpl
+++ b/gen/_template/router.tmpl
@@ -202,15 +202,18 @@ func (s *WebhookServer) Handler(webhookName string) http.Handler {
 		return
 	}
 {{- else }}
+	{{- with $routes := $r.Routes }}
+	if len(elem) == 0 {
+		{{ template "route_handle_request" $r }}
+		return
+	}
+	{{- end }}
 	{{- with $child := $r.StaticChildren }}
+		{{- if not $r.Routes }}
 		if len(elem) == 0 {
-			{{- with $routes := $r.Routes }}
-			{{ template "route_handle_request" $r }}
-			return
-			{{- else }}
 			break
-			{{- end }}
 		}
+		{{- end }}
 		switch elem[0] {
 		{{- range $c := $child }}
 		case {{ quote $c.Head }}: // Prefix: {{ quote $c.Prefix }}
@@ -221,7 +224,7 @@ func (s *WebhookServer) Handler(webhookName string) http.Handler {
 				break
 			}
 			{{ template "route_edge" router_elem $c $.ParameterIndex }}
-			
+
 			elem = origElem
 		{{- end }}
 		}

